core/configuredivecomputerthreads.cpp

```diff
@@ -360,16 +361,36 @@ static dc_status_t write_suunto_vyper_settings(dc_device_t *device, DeviceDetail
 	EMIT_PROGRESS();
 	return rc;
 }
+#undef EMIT_PROGRESS
+
+#define EMIT_PROGRESS()	do { \
+		progress.current++; \
+		progress_cb(device_data.device, DC_EVENT_PROGRESS, &progress, userdata); \
+	} while (0)
+
+static dc_status_t ostc3_sync_time(device_data_t &data)
+{
+	dc_status_t rc = DC_STATUS_SUCCESS;
+
+	if (data.sync_time) {
+		dc_datetime_t now;
+		dc_datetime_localtime(&now, dc_datetime_now());
+
+		rc = dc_device_timesync(data.device, &now);
+	}
+
+	return rc;
+}
 
-static dc_status_t read_ostc4_settings(dc_device_t *device, DeviceDetails &deviceDetails, dc_event_callback_t progress_cb, void *userdata)
+static dc_status_t read_ostc4_settings(device_data_t &device_data, DeviceDetails &deviceDetails, dc_event_callback_t progress_cb, void *userdata)
 {
 	// This code is really similar to the OSTC3 code, but there are minor
 	// differences in what the data means, and how to communicate with the
 	// device. If anyone can find a good way to harmonize the two, be my guest.
 	dc_status_t rc = DC_STATUS_SUCCESS;
 	dc_event_progress_t progress;
 	progress.current = 0;
-	progress.maximum = 47;
+	progress.maximum = 48;
```

Comment:
I think this value isn't correct. Either it was wrong before, or it's wrong after this.

There's two READ_SETTING calls added, which calls EMIT_PROGRESS , and two manual calls to EMIT_PROGRESS, so I'd guess it should be 51.

I had some prototype code to generate these counters, but that was ugly and it was nested in the old qmake build system, so it died in the cmake migration.

Reply:
Yes, I wasn't aware of `progress.maximum` when I added in the extra settings, and only came across it when changing the time synchronisation later on. Updating now.